src/libraries/UniswapV3Lib.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.21;
+
+import { IERC20 } from "openzeppelin-contracts/contracts/interfaces/IERC20.sol";
+import { IERC20Metadata } from "openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import { ERC20Lib } from "./ERC20Lib.sol";
+
+import { IALMProxy }                                                    from "../interfaces/IALMProxy.sol";
+import { IRateLimits }                                                  from "../interfaces/IRateLimits.sol";
+import { ISwapRouter, IUniswapV3PoolLike, INonfungiblePositionManager } from "../interfaces/UniswapV3Interfaces.sol";
+
+import { FullMath }   from "lib/dss-allocator/src/funnels/uniV3/FullMath.sol";
+import { UniV3Utils } from "lib/dss-allocator/test/funnels/UniV3Utils.sol";
+import { TickMath }   from "lib/dss-allocator/src/funnels/uniV3/TickMath.sol";
+
+import { RateLimitHelpers } from "../RateLimitHelpers.sol";
+
+import { UniswapV3OracleLib } from "./UniswapV3OracleLib.sol";
+
+library UniswapV3Lib {
+    uint24 public constant MAX_TICK_DELTA = 887272; // From https://github.com/sky-ecosystem/dss-allocator/blob/dev/src/funnels/uniV3/TickMath.sol#L15
+
+    /**********************************************************************************************/
+    /*** Structs                                                                                ***/
+    /**********************************************************************************************/
+    struct UniswapV3PoolParams {
+        uint24 swapMaxTickDelta;
+        uint32 swapTwapSecondsAgo;
+    }
+
+    struct UniV3Context {
+        IALMProxy   proxy;
+        IRateLimits rateLimits;
+        bytes32     rateLimitId;
+        address     pool;
+        uint256     deadline;
+    }
+
+    struct SwapParams {
+        UniswapV3PoolParams poolParams;
+        ISwapRouter         router;
+        address             tokenIn;
+        uint256             amountIn;
+        uint256             minAmountOut;
+        uint24              tickDelta; // The maximum that the tick can move by after completing the swap; type(uint24).max for no limit
```

Comment:
"No limit" will not work as it is checked against `swapMaxTickDelta` which has to be less than `MAX_TICK_DELTA`